assets/src/adaptivity/rules-engine.ts

```diff
@@ -304,11 +303,21 @@ export const getReferencedKeysInConditions = (conditions: any) => {
     ) {
       // value could have more than one reference inside it
       const exprs = extractAllExpressionsFromText(condition.value);
```

Comment:
The extracAllExpressionFromText was not extracting all the required variable for the expression - 

`{roundSF((sqrt(((pow(({q:1499808158045:412|stage.mass_1.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1526944957080:877|stage.mass_2.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1526945196262:881|stage.mass_3.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015057571:986|stage.mass_4.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015061771:990|stage.mass_5.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015064687:994|stage.mass_6.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015084238:998|stage.mass_7.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015086440:1002|stage.mass_8.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2))+(pow(({q:1527015088351:1006|stage.mass_9.Current Display Value}-{q:1499809254915:433|stage.avg_mass.value}),2)))/8)),1)}`